python/packages/ai/teams/streaming/streaming_response.py

```diff
@@ -111,6 +112,24 @@ def updates_sent(self) -> int:
         """
         return self._next_sequence - 1
 
+    def set_citations(self, citations: List[Citation]) -> None:
+        if len(citations) > 0:
+            if not self._citations:
+                self._citations = []
+            curr_pos = len(self._citations)
+
+            for citation in citations:
+                self._citations.append(
+                    ClientCitation(
+                        position=f"{curr_pos + 1}",
+                        appearance=Appearance(
+                            name=citation.title or f"Document {curr_pos + 1}",
+                            abstract=snippet(citation.content, 477),
```

Comment:
NIT: Should magic numbers be at the top of the doc?

Reply:
I kept them here to match our other languages / how we do it for the regular SAY commands, but we could update if we'd like